eva_submission/eload_brokering.py

```diff
@@ -0,0 +1,194 @@
+                ftps.storbinary('STOR %s' % vcf_file_index_name, open_file)
+
+    def upload_xml_files_to_ena(self, submission_file, project_file, analysis_file):
+        response = requests.post(
+            cfg.query('ena', 'submit_url'),
+            files=dict(SUBMISSION=submission_file, PROJECT=project_file, ANALYSIS=analysis_file)
+        )
+        # TODO: complete the parsing of the response
+
+    def upload_to_bioSamples(self, metadata_spreadsheet):
+        sample_tab_submitter = SampleMetadataSubmitter(metadata_spreadsheet)
+        sample_name_to_accession = sample_tab_submitter.submit_to_bioSamples()
+        self.eload_cfg['brokering']['Biosamples'] = sample_name_to_accession
+
+    def _run_brokering_prep_workflow(self):
+        output_dir = self.create_temp_output_directory()
+        brokering_config = {
+            'vcf_files': self.eload_cfg['validation']['valid']['vcf_files'],
+            'output_dir': output_dir,
+            'executable': cfg['executable']
+        }
+        # run the validation
+        brokering_confg_file = os.path.join(self.eload_dir, 'brokering_confg_file.yaml')
+        with open(brokering_confg_file, 'w') as open_file:
+            yaml.safe_dump(brokering_config, open_file)
+        validation_script = os.path.join(os.path.dirname(os.path.dirname(__file__)), 'nextflow', 'prepare_brokering.nf')
+        try:
+            command_utils.run_command_with_output(
+                'Start Nextflow brokering preparation process',
+                ' '.join((
+                    cfg['executable']['nextflow'], validation_script,
+                    '-params-file', brokering_confg_file,
+                    '-work-dir', output_dir
+                ))
+            )
+        except subprocess.CalledProcessError as e:
+            self.error('Nextflow pipeline failed: aborting brokering')
+            raise e
+        return output_dir
+
+    def _collect_brokering_worklflow_results(self, output_dir):
+        # Collect information from the output and summarise in the config
+        nextflow_vcf_output = os.path.join(output_dir, 'output')
+        for vcf_file in self.eload_cfg.query('validation', 'valid', 'vcf_files'):
+            vcf_file_name = os.path.basename(vcf_file)
+            if not vcf_file_name.endswith('.gz'):
+                vcf_file_name = vcf_file_name + '.gz'
+
+            ovcf_file = os.path.join(self._get_dir('ena'), vcf_file_name)
+            os.rename(os.path.join(nextflow_vcf_output, vcf_file_name), ovcf_file)
+            os.rename(os.path.join(output_dir, vcf_file_name) + '.md5', ovcf_file + '.md5')
+
+            ifile = os.path.join(output_dir, vcf_file_name + '.csi')
+            o_index_file = os.path.join(self._get_dir('ena'), vcf_file_name + '.csi')
+            os.rename(ifile, o_index_file)
+            os.rename(ifile + '.md5', o_index_file + '.md5')
+
+            self.eload_cfg.set('brokering', 'vcf_files', ovcf_file, value={
```

Comment:
Is **ovcf** the original VCF file?

Reply:
That was suppose to mean `output_vcf` but was obviously not clear. I've clarified.